core/sql/generator/GenRelMisc.cpp

```diff
@@ -3071,7 +3074,16 @@ short Sort::generateTdb(Generator * generator,
   sort_options->scratchFreeSpaceThresholdPct() = threshold;
   sort_options->sortMaxHeapSize() = (short)getDefault(SORT_MAX_HEAP_SIZE_MB);
   sort_options->mergeBufferUnit() = (short)getDefault(SORT_MERGE_BUFFER_UNIT_56KB);
+  
   sort_options->scratchIOBlockSize() = (Int32)getDefault(SCRATCH_IO_BLOCKSIZE_SORT);
+  if(sortRecLen >= sort_options->scratchIOBlockSize())
+  {
+    Int32 maxScratchIOBlockSize = (Int32)getDefault(SCRATCH_IO_BLOCKSIZE_SORT_MAX);
+    GenAssert(sortRecLen <= maxScratchIOBlockSize, 
+         "sortRecLen is greater than SCRATCH_IO_BLOCKSIZE_SORT_MAX");
+    sort_options->scratchIOBlockSize() = MINOF(sortRecLen * 128, maxScratchIOBlockSize);
```

Comment:
Not sure I understand. It looks like this code sizes the scratch IO block size to have at most 128 records (rather than at least).